### PR TITLE
Add "New" badge to Key Metrics dashboard widgets.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue:

- #7376

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/sass/components/key-metrics/_googlesitekit-key-metrics-widgets.scss [hunk 1]
```diff
@@ -28,6 +28,16 @@
...
 			}
 		}
+
+		.googlesitekit-widget-area-header__title {
+
+			align-items: center;
+			display: flex;
+
+			.googlesitekit-new-badge {
+				margin-left: 8px;
```
Comment (hussain-t) [inline assets/sass/components/key-metrics/_googlesitekit-key-metrics-widgets.scss:38]: The gap between the `Key metrics` title and the `New` badge seems lesser than the Figma design. Can we make it `10px` or `12px`?

**Figma design**

Screenshot 2023-08-24 at 10 03 40 PM

**Current implementation**

Screenshot 2023-08-24 at 10 03 21 PM

**Review by hussain-t — CHANGES_REQUESTED** [review]
Thanks, @tofumatt. This looks good. Left a nitpicky comment regarding the gap between the `Key metrics` title and the `New` badge. Please take a look at it.

File: assets/sass/components/key-metrics/_googlesitekit-key-metrics-widgets.scss (hunk 1 above)
Comment (tofumatt) [reply assets/sass/components/key-metrics/_googlesitekit-key-metrics-widgets.scss]: Good call, I've made the change 👍🏻

**Review by hussain-t — APPROVED** [review]
Thanks, @tofumatt. LGTM 👍

**Review by techanvil — APPROVED** [review]
LGTM!

**Pull request merged**